kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -651,45 +626,54 @@ class ResidualBasedBlockBuilderAndSolverWithConstraints
     {
         KRATOS_TRY
         int slave_count = 0;
-        for (auto &slave_equation_id : rSlaveEquationIdVector)
+        for (auto slave_equation_id : rSlaveEquationIdVector)
         {
+            /*int master_count = 0;
+            auto global_constraint = mGlobalMasterSlaveConstraints.find(slave_equation_id);
+            if (global_constraint == mGlobalMasterSlaveConstraints.end())
+            {
+                mGlobalMasterSlaveConstraints.insert(mGlobalMasterSlaveConstraints.begin(), Kratos::make_shared<AuxiliaryGlobalMasterSlaveConstraintType>(slave_equation_id));
+                global_constraint = mGlobalMasterSlaveConstraints.find(slave_equation_id);
+            }*/
             int master_count = 0;
-            auto global_constraint = mGlobalMasterSlaveRelations.find(slave_equation_id);
-            if (global_constraint == mGlobalMasterSlaveRelations.end())
+            if (mGlobalMasterSlaveConstraints.count(slave_equation_id) == 0 )
             {
-                mGlobalMasterSlaveRelations.insert(mGlobalMasterSlaveRelations.begin(), Kratos::make_shared<AuxilaryGlobalMasterSlaveRelationType>(slave_equation_id));
-                global_constraint = mGlobalMasterSlaveRelations.find(slave_equation_id);
+                mGlobalMasterSlaveConstraints[slave_equation_id] = Kratos::make_unique<AuxiliaryGlobalMasterSlaveConstraintType>(slave_equation_id);
             }
-            for (auto &master_equation_id : rMasterEquationIdVector)
+
+            auto& global_constraint = mGlobalMasterSlaveConstraints[slave_equation_id];
+            for (auto master_equation_id : rMasterEquationIdVector)
             {
                 global_constraint->AddMaster(master_equation_id, rRelationMatrix(slave_count, master_count));
                 master_count++;
             }
             slave_count++;
         }
-        KRATOS_CATCH("ResidualBasedBlockBuilderAndSolverWithConstraints::AssembleSlaves failed ..");
+        KRATOS_CATCH("ResidualBasedBlockBuilderAndSolverWithConstraints::AssembleSlaves failed ...");
```

Comment:
@KratosMultiphysics/technical-committee  this is a reminder for myself. I wonder if it is expensive to have long messages in the catch